js/parts/Series.js

```diff
@@ -3095,8 +3095,10 @@ H.Series = H.seriesType('line', null, { // base series options
                 // Search for the same X in the existing data set
                 pointIndex = H.inArray(x, this.xData, lastIndex);
 
-                // Matching X not found, add point (but later)
-                if (pointIndex === -1) {
+                // Matching X not found
+                // or used already due to ununique x values (#8995),
+                // add point (but later)
+                if (pointIndex === -1 || oldData[pointIndex].touched) {
```

Comment:
@TorsteinHonsi 
`|| oldData[pointIndex].touched` is for series that do not have data sorted - it avoids using the same point again, so a new point is added. This probably could be improved by keep looking for next point with the same x value that was not updated in current `setData` run, but I was unable to find a solution that would be light and fast.